controller/api/responder.go

```diff
@@ -67,15 +69,26 @@ func (responder *ResponderImpl) Respond(data interface{}, httpStatus int) {
 
 func (responder *ResponderImpl) RespondWithProducer(producer runtime.Producer, data interface{}, httpStatus int) {
 	w := responder.rc.GetResponseWriter()
+	buff := &bytes.Buffer{}
+	err := producer.Produce(buff, data)
+
+	if err != nil {
+		pfxlog.Logger().WithError(err).
+			WithField("requestId", responder.rc.GetId()).
+			WithField("path", responder.rc.GetRequest().URL.Path).
+			Debug("could not respond, producer errored, possible missing response content")
```

Comment:
should this be at error for now?

Reply:
if it fails to produce bytes we should set error on response, right?

Reply:
It gets weird in here. We are at a very low level and the probability that the producer will error is low.

How should I respond if the producer that producer the right content format fails to format? Should I call the producer again with the error record? Seems awkward if the producer is erroring.

I can send an error code and a plain/text body w/ the error message. Attempting to format the output w/ the same producer is dubious.